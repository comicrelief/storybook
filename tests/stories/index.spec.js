import initStoryshots, { snapshotWithOptions } from '@storybook/addon-storyshots';
import { mockRandomForEach } from 'jest-mock-random';

describe('SchoolsLookup', () => {
  // Math.random() is used to generate a prop value
  // Here we mock Math.random(), to make sure return is the same everytime we run the test
  // That way, existing snapshots wil always match render result
  mockRandomForEach([0.1]);

  // Snapshots set all refs to null
  // So any ref needs to be mocked manually
  const createNodeMock = (el) => {
    switch (el.type) {
      case 'input':
      case 'div':
        return document.createElement(el.type);

      default:
        return null;
    }
  };

  // test SchoolsLookUp layout at it's lifecycle different states
  initStoryshots({
    storyKindRegex: /^SchoolsLookUp$/,
    test: snapshotWithOptions({ createNodeMock }),
  });
});
