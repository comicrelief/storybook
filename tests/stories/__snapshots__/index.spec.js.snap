// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchoolsLookup Storyshots SchoolsLookUp EDCO selected school 1`] = `
<div
  className="SchoolsLookUp"
>
  <p
    className="schoolsLookUp-title"
  >
    <label
      htmlFor="schoolsLookUp"
    >
      Enter the name or postcode of your school or nursery
    </label>
  </p>
  <div>
    <div
      className="schoolDetails"
    >
      <div
        className=""
      >
        <p>
          School xyz
        </p>
      </div>
      <div
        className=""
      >
        <p>
          dummy address line 1
        </p>
      </div>
      <div
        className=""
      >
        <p>
          dummy town
        </p>
      </div>
      <div
        className=""
      >
        <p>
          dummy postcode
        </p>
      </div>
    </div>
    <button
      className="SchoolsLookUp-link"
      name="edit"
      onClick={[Function]}
    >
      Edit
    </button>
  </div>
</div>
`;

exports[`SchoolsLookup Storyshots SchoolsLookUp manual fields errors 1`] = `
<div
  className="SchoolsLookUp"
>
  <p
    className="schoolsLookUp-title"
  >
    <label
      htmlFor="schoolsLookUp"
    >
      Enter the name or postcode of your school or nursery
    </label>
  </p>
  <div
    className="schoolsLookUp-search"
  >
    <div
      className="rbt clearfix open schoolsLookUpForm"
      style={
        Object {
          "position": "relative",
        }
      }
      tabIndex={-1}
    >
      <div
        className="rbt-input-hint-container"
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <input
          aria-activedescendant=""
          aria-autocomplete="both"
          aria-expanded={false}
          aria-haspopup="listbox"
          aria-owns="rbt-menu-3lllllllllm"
          autoComplete="nope"
          className="rbt-input-main form-control rbt-input  input-lg form-control-lg"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="Type to start search"
          role="combobox"
          type="text"
          value=""
        />
        <input
          aria-hidden={true}
          className="rbt-input-hint"
          style={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "boxShadow": "none",
              "color": "rgba(0, 0, 0, 0.35)",
              "left": 0,
              "pointerEvents": "none",
              "position": "absolute",
              "top": 0,
            }
          }
          tabIndex={-1}
          value=""
        />
      </div>
      <div
        aria-atomic={true}
        aria-live="polite"
        className="sr-only rbt-sr-status"
        role="status"
      >
        0 selections
      </div>
    </div>
  </div>
  <div
    className="SchoolsLookUp-maunal"
  >
    <p>
      Or enter details manually
    </p>
    <div
      className="schoolDetails"
    >
      <div
        className="validation__wrapper"
      >
        <div>
          <label
            className="required"
            htmlFor="establishmentNameValue"
          >
            Establishment name
          </label>
          <input
            id="establishmentNameValue"
            name="establishmentNameValue"
            onBlur={[Function]}
            onChange={[Function]}
            required={true}
            type="text"
            value=""
          />
          <div
            className="validation__message"
          >
            <span>
              establishmentName error message
            </span>
          </div>
        </div>
      </div>
      <div
        className="validation__wrapper"
      >
        <div>
          <label
            className="required"
            htmlFor="address1Value"
          >
            Address line 1
          </label>
          <input
            id="address1Value"
            name="address1Value"
            onBlur={[Function]}
            onChange={[Function]}
            required={true}
            type="text"
            value=""
          />
          <div
            className="validation__message"
          >
            <span>
              address1 error message
            </span>
          </div>
        </div>
      </div>
      <div
        className="validation__wrapper"
      >
        <div>
          <label
            className=""
            htmlFor="address2Value"
          >
            Address line 2
          </label>
          <input
            id="address2Value"
            name="address2Value"
            onBlur={[Function]}
            onChange={[Function]}
            required={false}
            type="text"
            value=""
          />
          <div
            className="validation__message"
          >
            <span>
              address2 error message
            </span>
          </div>
        </div>
      </div>
      <div
        className="validation__wrapper"
      >
        <div>
          <label
            className=""
            htmlFor="address3Value"
          >
            Address line 3
          </label>
          <input
            id="address3Value"
            name="address3Value"
            onBlur={[Function]}
            onChange={[Function]}
            required={false}
            type="text"
            value=""
          />
          <div
            className="validation__message"
          >
            <span>
              address3 error message
            </span>
          </div>
        </div>
      </div>
      <div
        className="validation__wrapper"
      >
        <div>
          <label
            className="required"
            htmlFor="townValue"
          >
            Town
          </label>
          <input
            id="townValue"
            name="townValue"
            onBlur={[Function]}
            onChange={[Function]}
            required={true}
            type="text"
            value=""
          />
          <div
            className="validation__message"
          >
            <span>
              town error message
            </span>
          </div>
        </div>
      </div>
      <div
        className="validation__wrapper"
      >
        <div>
          <label
            className="required"
            htmlFor="postcodeValue"
          >
            Postcode
          </label>
          <input
            id="postcodeValue"
            name="postcodeValue"
            onBlur={[Function]}
            onChange={[Function]}
            required={true}
            type="text"
            value=""
          />
          <div
            className="validation__message"
          >
            <span>
              postcode error message
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SchoolsLookup Storyshots SchoolsLookUp manually entered school 1`] = `
<div
  className="SchoolsLookUp"
>
  <p
    className="schoolsLookUp-title"
  >
    <label
      htmlFor="schoolsLookUp"
    >
      Enter the name or postcode of your school or nursery
    </label>
  </p>
  <div
    className="schoolsLookUp-search"
  >
    <div
      className="rbt clearfix open schoolsLookUpForm"
      style={
        Object {
          "position": "relative",
        }
      }
      tabIndex={-1}
    >
      <div
        className="rbt-input-hint-container"
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <input
          aria-activedescendant=""
          aria-autocomplete="both"
          aria-expanded={false}
          aria-haspopup="listbox"
          aria-owns="rbt-menu-3lllllllllm"
          autoComplete="nope"
          className="rbt-input-main form-control rbt-input  input-lg form-control-lg"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="Type to start search"
          role="combobox"
          type="text"
          value=""
        />
        <input
          aria-hidden={true}
          className="rbt-input-hint"
          style={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "boxShadow": "none",
              "color": "rgba(0, 0, 0, 0.35)",
              "left": 0,
              "pointerEvents": "none",
              "position": "absolute",
              "top": 0,
            }
          }
          tabIndex={-1}
          value=""
        />
      </div>
      <div
        aria-atomic={true}
        aria-live="polite"
        className="sr-only rbt-sr-status"
        role="status"
      >
        0 selections
      </div>
    </div>
  </div>
  <div
    className="SchoolsLookUp-maunal"
  >
    <p>
      Or enter details manually
    </p>
    <div
      className="schoolDetails"
    >
      <div
        className=""
      >
        <div>
          <label
            className="required"
            htmlFor="establishmentNameValue"
          >
            Establishment name
          </label>
          <input
            id="establishmentNameValue"
            name="establishmentNameValue"
            onBlur={[Function]}
            onChange={[Function]}
            required={true}
            type="text"
            value="School xyz"
          />
        </div>
      </div>
      <div
        className=""
      >
        <div>
          <label
            className="required"
            htmlFor="address1Value"
          >
            Address line 1
          </label>
          <input
            id="address1Value"
            name="address1Value"
            onBlur={[Function]}
            onChange={[Function]}
            required={true}
            type="text"
            value="dummy address line 1"
          />
        </div>
      </div>
      <div
        className=""
      >
        <div>
          <label
            className=""
            htmlFor="address2Value"
          >
            Address line 2
          </label>
          <input
            id="address2Value"
            name="address2Value"
            onBlur={[Function]}
            onChange={[Function]}
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className=""
      >
        <div>
          <label
            className=""
            htmlFor="address3Value"
          >
            Address line 3
          </label>
          <input
            id="address3Value"
            name="address3Value"
            onBlur={[Function]}
            onChange={[Function]}
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className=""
      >
        <div>
          <label
            className="required"
            htmlFor="townValue"
          >
            Town
          </label>
          <input
            id="townValue"
            name="townValue"
            onBlur={[Function]}
            onChange={[Function]}
            required={true}
            type="text"
            value="dummy town"
          />
        </div>
      </div>
      <div
        className=""
      >
        <div>
          <label
            className="required"
            htmlFor="postcodeValue"
          >
            Postcode
          </label>
          <input
            id="postcodeValue"
            name="postcodeValue"
            onBlur={[Function]}
            onChange={[Function]}
            required={true}
            type="text"
            value="dummy postcode"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SchoolsLookup Storyshots SchoolsLookUp nothing is selected 1`] = `
<div
  className="SchoolsLookUp"
>
  <p
    className="schoolsLookUp-title"
  >
    <label
      htmlFor="schoolsLookUp"
    >
      Enter the name or postcode of your school or nursery
    </label>
  </p>
  <div
    className="schoolsLookUp-search"
  >
    <div
      className="rbt clearfix open schoolsLookUpForm"
      style={
        Object {
          "position": "relative",
        }
      }
      tabIndex={-1}
    >
      <div
        className="rbt-input-hint-container"
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <input
          aria-activedescendant=""
          aria-autocomplete="both"
          aria-expanded={false}
          aria-haspopup="listbox"
          aria-owns="rbt-menu-3lllllllllm"
          autoComplete="nope"
          className="rbt-input-main form-control rbt-input  input-lg form-control-lg"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="Type to start search"
          role="combobox"
          type="text"
          value=""
        />
        <input
          aria-hidden={true}
          className="rbt-input-hint"
          style={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "boxShadow": "none",
              "color": "rgba(0, 0, 0, 0.35)",
              "left": 0,
              "pointerEvents": "none",
              "position": "absolute",
              "top": 0,
            }
          }
          tabIndex={-1}
          value=""
        />
      </div>
      <div
        aria-atomic={true}
        aria-live="polite"
        className="sr-only rbt-sr-status"
        role="status"
      >
        0 selections
      </div>
    </div>
  </div>
  <button
    className="SchoolsLookUp-link"
    name="enterManually"
    onClick={[Function]}
  >
    Or enter details manually
  </button>
</div>
`;
